File: Serene/Serene.Script/Administration/RolePermission/RolePermissionDialog.cs
namespace Serene.Administration
{
    using jQueryApi.UI.Widgets;
    using Serenity;
    using System;
    using System.Collections.Generic;
    using System.Html;
    using System.Runtime.CompilerServices;
    using System.Linq;

    [Imported]
    public class RolePermissionDialog : TemplatedDialog<RolePermissionDialogOptions>
    {
    }

    [Imported, Serializable]
    public class RolePermissionDialogOptions
    {
        public int RoleID { get; set; }
        public string Title { get; set; }
    }

    /*
    This class has been ported to TypeScript. See RoleDialog.ts
    Code below is only for reference purposes who wants to use Saltaralle 

    public class RolePermissionDialog : TemplatedDialog<RolePermissionDialogOptions>
    {
        private PermissionCheckEditor permissions;
        
        public RolePermissionDialog(RolePermissionDialogOptions opt)
            : base(opt)
        {
            permissions = new PermissionCheckEditor(this.ById("Permissions"), 
                new PermissionCheckEditorOptions { ShowRevoke = false });

            RolePermissionService.List(new RolePermissionListRequest
            {
                RoleID = options.RoleID,
                Module = null,
                Submodule = null
            }, response =>
            {
                permissions.Value = response.Entities.Select(x => new UserPermissionRow { PermissionKey = x }).ToList();
            });
        }

        protected override DialogOptions GetDialogOptions()
        {
            var opt = base.GetDialogOptions();
            opt.Buttons = new List<DialogButton>
            {
                new DialogButton {
                    Text = Q.Text("Dialogs.OkButton"),
                    Click = delegate
                    {
                        RolePermissionService.Update(new RolePermissionUpdateRequest
                        {
                            RoleID = options.RoleID,
                            Permissions = permissions.Value.Select(x => x.PermissionKey).ToList(),
                            Module = null,
                            Submodule = null
                        }, response => {
                            DialogClose();
                            Window.SetTimeout(delegate()
                            {
                                Q.NotifySuccess(Q.Text("Site.RolePermissionDialog.SaveSuccess"));
                            }, 0);
                        });
                    },
                },
                new DialogButton {
                    Text = Q.Text("Dialogs.CancelButton"),
                    Click = DialogClose
                }
            };
            opt.Title = String.Format(Q.Text("Site.RolePermissionDialog.DialogTitle"), options.Title);
            return opt;
        }

        protected override string GetTemplate()
        {
            return "<div id='~_Permissions'></div>";
        }
    }
    */
}
 

File: Serene/Serene.Web/Modules/Administration/RolePermission/RolePermissionDialog.ts
namespace Serene.Administration {

    @Serenity.Decorators.registerClass()
    export class RolePermissionDialog extends Serenity.TemplatedDialog<RolePermissionDialogOptions> {

        private permissions: PermissionCheckEditor;

        constructor(opt: RolePermissionDialogOptions) {
            super(opt);

            this.permissions = new PermissionCheckEditor(this.byId('Permissions'), {
                showRevoke: false
            });

            RolePermissionService.List({
                RoleID: this.options.roleID,
                Module: null,
                Submodule: null
            }, response => {
                this.permissions.set_value(response.Entities.map(x => (<UserPermissionRow>{ PermissionKey: x })));
            });
        }

        protected getDialogOptions(): JQueryUI.DialogOptions {
            let opt = super.getDialogOptions();

            opt.buttons = [
                {
                    text: Q.text('Dialogs.OkButton'),
                    click: e => {
                        RolePermissionService.Update({
                            RoleID: this.options.roleID,
                            Permissions: this.permissions.get_value().map(x => x.PermissionKey),
                            Module: null,
                            Submodule: null
                        }, response => {
                            this.dialogClose();
                            window.setTimeout(() => Q.notifySuccess(Q.text('Site.RolePermissionDialog.SaveSuccess')), 0);
                        });
                    }
                }, {
                    text: Q.text('Dialogs.CancelButton'),
                    click: () => this.dialogClose()
                }];

            opt.title = ss.formatString(Q.text('Site.RolePermissionDialog.DialogTitle'),
                this.options.title);

            return opt;
        }

        protected getTemplate(): string {
            return '<div id="~_Permissions"></div>';
        }
    }

    export interface RolePermissionDialogOptions {
        roleID?: number;
        title?: string;
    }
}